pkg/metrics/backends/prometheus/prometheus.go

```diff
@@ -93,3 +97,51 @@ func (b *Backend) QueryInstant(ctx context.Context, query string, ts time.Time)
 		return nil, errors.Errorf("unknown supported type: '%q'", v)
 	}
 }
+
+// getAlertmanagers returns Alertmanager clients that is associated with the backend.
+func (b *Backend) getAlertmanagers(ctx context.Context) ([]*amclient.Alertmanager, error) {
+	level.Debug(b.logger).Log("msg", "Get active Alertmanagers")
+	amResult, err := b.prometheus.AlertManagers(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var ams []*amclient.Alertmanager
+	// Get only active Alertmanager instances.
+	for _, a := range amResult.Active {
+		u, _ := url.Parse(a.URL)
+		level.Debug(b.logger).Log("msg", "Setup Alertmanager client", "alertmanager", u.Host)
+		// NOTE(kiennt): This is Alertmanager API Ver 2 client.
+		// https://github.com/prometheus/alertmanager/blob/master/api/v2/client/alertmanager_client.go
+		amCli := amclient.NewHTTPClientWithConfig(nil, &amclient.TransportConfig{
+			Host:     u.Host,
+			BasePath: "/api/v2/", // hardcode here
+			Schemes:  amclient.DefaultSchemes,
+		})
+		ams = append(ams, amCli)
+	}
+	return ams, nil
+}
+
+// GetAlertmanagerSilences returns silences in Alertmanagers.
+func (b *Backend) GetAlertManagerSilences(ctx context.Context, filter []string) (map[string]ammodels.Silence, error) {
+	level.Debug(b.logger).Log("msg", "Get Alertmanager silences")
```

Comment:
Debug log are overloaded. throw it at https://github.com/vCloud-DFTBA/faythe/pull/35/files#diff-c8a3dc6adf2ccc913d7cc34f8165bac5f8757f5924628e23da834e404ac4937eR143 to notify valid silences.